src/components/src/main/java/org/apache/jmeter/visualizers/backend/influxdb/HttpMetricsSender.java

```diff
@@ -150,73 +156,69 @@ public void addMetric(String mesurement, String tag, String field) {
-                    @Override
-                    public void completed(final HttpResponse response) {
-                        int code = response.getStatusLine().getStatusCode();
-                        /*
-                         * HTTP response summary 2xx: If your write request received
-                         * HTTP 204 No Content, it was a success! 4xx: InfluxDB
-                         * could not understand the request. 5xx: The system is
-                         * overloaded or significantly impaired.
-                         */
-                        if (MetricUtils.isSuccessCode(code)) {
-                            if(log.isDebugEnabled()) {
-                                log.debug("Success, number of metrics written: {}", copyMetrics.size());
-                            }
-                        } else {
-                            log.error("Error writing metrics to influxDB Url: {}, responseCode: {}, responseBody: {}", url, code, getBody(response));
+            httpRequest.setEntity(new StringEntity(sb.toString(), StandardCharsets.UTF_8));
+            lastRequest = httpClient.execute(httpRequest, new FutureCallback<HttpResponse>() {
+                @Override
+                public void completed(final HttpResponse response) {
+                    int code = response.getStatusLine().getStatusCode();
+                    /*
+                     * If your write request received HTTP
+                     * 204 No Content: it was a success!
+                     * 4xx: InfluxDB could not understand the request.
+                     * 5xx: The system is overloaded or significantly impaired.
+                     */
+                    if (MetricUtils.isSuccessCode(code)) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Success, number of metrics written: {}", copyMetrics.size());
                         }
+                    } else {
+                        log.error("Error writing metrics to influxDB Url: {}, responseCode: {}, responseBody: {}", url, code, getBody(response));
                     }
-                    @Override
-                    public void failed(final Exception ex) {
-                        log.error("failed to send data to influxDB server : {}", ex.getMessage());
-                    }
-                    @Override
-                    public void cancelled() {
-                        log.warn("Request to influxDB server was cancelled");
-                    }
-                });
-            }catch (URISyntaxException ex ) {
-                log.error(ex.getMessage());
-            } finally {
-                // We drop metrics in all cases
-                copyMetrics.clear();
-            }
+                }
+
+                @Override
+                public void failed(final Exception ex) {
+                    log.error("failed to send data to influxDB server.", ex);
```

Comment:
I think it was intentional not to write the full stacktrace for performance reasons

Reply:
Oh I see. If that's the case, then either this maybe shouldn't be logged as an error, or we could add a debug log with the full stacktrace in?

Reply:
This happens *asynchronous* to the main processing, so the performance does not matters much here.
The code relates to error processing that should not happen often.
The stacktrace would likely help to analyze and fix the issue, so having a stacktrace is very great.

I think we should go with @ham1 suggestion here. If somebody thinks we should really improve performance for the case when `failed` is executed very often, then we should likely put some throttling here (e.g. log no more than 5 stacktraces per minute), however that would be completely different issue.

I believe the proposed change helps a lot for end users to understand the nature of the failure.
If they face "too much stacktraces logged" and they are fine with errors, then they could turn off the logger for the particular class in question.

Reply:
It will also hit performances by flooding log file with Stacktraces. So it's also about usability. And it can end up filling the disk and breaking the test. I faced such case and that's why I reverted to removing the exception stacktrace logging. Y

Reply:
> It will also hit performances by flooding log file with Stacktraces

Do you consider it to be the major use-case?

Reply:
The question is here, do you want to break a load test because of temporary communication issues or unavailability of InfluxDB ? 
What do you mean by the major use case ?

Reply:
>And it can end up filling the disk and breaking the tes

Just in case: `new Throwable()` populates the stacktrace, thus it impacts performance as well.

The code above has no commets that clarify that `ex.getMessage()` was used intentionally, thus I fully agree with @ham1.

Is "errors from Influx" happening really often? I just thought it means there's a misconfiguration, so the test have to be re-run in any case.

Reply:
> temporary communication issues or unavailability of InfluxDB ?

Temporal unavailability should not cause test breakage.
However, it should not produce lots of errors as well.

Note: if JMeter can recover from temporal InfluxDB unavailability (==it can re-send the data), then the message must not be logged as `error` at all.

Just in case:
`error` means "the system is broken, dear system administrator, please heal it"
`warn` means "there's something wrong, but it can live with it. Dear administrator, you might want to check the system"

Reply:
On the other hand, if `temporal unavailability` means `loss of datapoints sent to InfluxDB`, then it might easily make test invalid, as the results will look "all good" while the most critical data points (the ones that are related to unavailability, thus "bad performance") could be just missing.

Reply:
Which of the following sounds best to you:
1. log.error(ex.message); log.debug(ex)
2. log.warn(ex.message); log.debug(ex)
3. log.warn(ex)
4. log.error(ex)

Reply:
"migrate everything to junit5" :)

Reply:
Frankly speaking, I don't know if `failure` means the loss of the data or not.
If the data is lost, then it should be `log.error(message that explains the loss, ex);`.
However as @pmouawad  says there might be lots of exceptions, so there should be throttling that prevents logging the stacktrace more often than once per minute.

Reply:
I think it most likely does mean loss of data sent to InfluxDB, because we clear the metrics after trying to send them.